src/Compilers/CSharp/Portable/Errors/CSharpDiagnosticFormatter.cs

```diff
@@ -1,15 +1,29 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System.Globalization;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics.CSharp;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     public class CSharpDiagnosticFormatter : DiagnosticFormatter
     {
-        internal CSharpDiagnosticFormatter()
+        private static readonly string CompilerAnalyzer = typeof(CSharpCompilerDiagnosticAnalyzer).ToString();
+
+        internal CSharpDiagnosticFormatter(bool displayAnalyzer) :
+            base(displayAnalyzer)
         {
         }
 
-        public new static readonly CSharpDiagnosticFormatter Instance = new CSharpDiagnosticFormatter();
+        protected override string GetAnalyzerIdentity(DiagnosticAnalyzer analyzer)
+        {
+            if (analyzer == null)
+            {
+                return CompilerAnalyzer;
```

Comment:
ya, I thought about that. that might be just simpler. I first tried not to make compiler one any special than others. let's see as I finish work around this which way is simpler.